client/components/language-picker/modal.jsx

```diff
@@ -417,7 +415,33 @@ export class LanguagePickerModal extends PureComponent {
 		this.handleClose();
 	};
 
-	handleClose = () => {
+	recordLanguagePickedEvent = ( isClosingWithoutSelection ) => {
+		const { selectedLanguageSlug, search } = this.state;
+
+		if (
+			isClosingWithoutSelection ||
+			this.initiallySelectedLanguageSlug === selectedLanguageSlug
+		) {
+			// we don't record an event if the language wasn't changed
+			return;
+		}
+
+		/**
+		 * We match specifically against the default value of `false`
+		 * rather than treating this falsy as `search` may be an empty
+		 * string if the user searches and then closes the search bar.
+		 * We'll naïvely assume that if someone searched *at all* that
+		 * we should record them as having used the search to find the
+		 * language they want. It's the simplest assumption for the
+		 * moment, and I'm not sure there's a better solution that
+		 * wouldn't be overkill given what we're trying to record here.
+		 */
+		const searched = false !== search;
```

Comment:
What if we have a state prop `searched` that's initially `false`, and we change it to `true` once `handleSearch` is called at least once?